exporters/trace/jaeger/src/main/java/io/opencensus/exporter/trace/jaeger/JaegerExporterHandler.java

```diff
@@ -0,0 +1,273 @@
+    this.sender = checkNotNull(sender, "Jaeger sender must NOT be null.");
+    this.process = checkNotNull(process, "Process sending traces must NOT be null.");
+  }
+
+  @Override
+  public void export(final Collection<SpanData> spanDataList) {
+    final List<Span> spans = spanDataToJaegerThriftSpans(spanDataList);
+    logger.log(Level.INFO, "About to export " + spans.size() + " traces to Jaeger");
+    exportToJaeger(spans);
+  }
+
+  private List<Span> spanDataToJaegerThriftSpans(final Collection<SpanData> spanDataList) {
+    final List<Span> spans = Lists.newArrayListWithExpectedSize(spanDataList.size());
+    for (final SpanData spanData : spanDataList) {
+      spans.add(spanDataToJaegerThriftSpan(spanData));
+    }
+    return spans;
+  }
+
+  private void exportToJaeger(final List<Span> spans) {
+    try {
+      sender.send(process, spans);
+    } catch (TException e) {
+      tracer
+          .getCurrentSpan()
+          .setStatus(
+              Status.UNKNOWN.withDescription(
+                  e.getMessage() == null ? e.getClass().getSimpleName() : e.getMessage()));
+      throw new RuntimeException(e); // TODO: should we instead do drop metrics?
+    }
+  }
+
+  private Span spanDataToJaegerThriftSpan(final SpanData spanData) {
+    final long startTimeInMicros = timestampToMicros(spanData.getStartTimestamp());
+    final long endTimeInMicros = timestampToMicros(spanData.getEndTimestamp());
+
+    final SpanContext context = spanData.getContext();
+    copyToBuffer(context.getTraceId());
+
+    return new com.uber.jaeger.thriftjava.Span(
+            traceIdLow(),
+            traceIdHigh(),
+            spanIdToLong(context.getSpanId()),
+            spanIdToLong(spanData.getParentSpanId()),
+            spanData.getName(),
+            optionsToFlags(context.getTraceOptions()),
+            startTimeInMicros,
+            endTimeInMicros - startTimeInMicros)
+        .setReferences(linksToReferences(spanData.getLinks().getLinks()))
+        .setTags(attributesToTags(spanData.getAttributes().getAttributeMap()))
+        .setLogs(annotationEventsToLogs(spanData.getAnnotations().getEvents()));
+  }
+
+  private void copyToBuffer(final TraceId traceId) {
+    // Attempt to minimise allocations, since TraceId#getBytes currently creates a defensive copy:
+    traceId.copyBytesTo(traceIdBuffer, 0);
+  }
+
+  private long traceIdHigh() {
+    return Longs.fromBytes(
+        traceIdBuffer[0],
+        traceIdBuffer[1],
+        traceIdBuffer[2],
+        traceIdBuffer[3],
+        traceIdBuffer[4],
+        traceIdBuffer[5],
+        traceIdBuffer[6],
+        traceIdBuffer[7]);
+  }
+
+  private long traceIdLow() {
+    return Longs.fromBytes(
+        traceIdBuffer[8],
+        traceIdBuffer[9],
+        traceIdBuffer[10],
+        traceIdBuffer[11],
+        traceIdBuffer[12],
+        traceIdBuffer[13],
+        traceIdBuffer[14],
+        traceIdBuffer[15]);
+  }
+
+  private long spanIdToLong(final @Nullable SpanId traceId) {
```

Comment:
The input argument is named `traceId`, which I think is misleading.

Reply:
Sorry, bad copy-pasting, consider it fixed!